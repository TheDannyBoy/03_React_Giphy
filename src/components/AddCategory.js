import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = evt => {
    setInputValue(evt.target.value);
  };

  const handleSubmit = evt => {
    evt.preventDefault();
    setCategories(categories => ([ inputValue, ...categories ]));
    setInputValue('');
  };

  return (<form onSubmit={handleSubmit}>
    <h1>{ inputValue }</h1>
    <input
      type='text'
      value={inputValue}
      onChange={handleInputChange}>
    </input>
  </form>)
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}
