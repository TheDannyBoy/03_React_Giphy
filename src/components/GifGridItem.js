import React from 'react'

export const GifGridItem = ({ title, url }) => {
  const classes = ['card'];
  return (
    <div className={classes.join(' ')}>
      <img src={url} alt={title} />
      <p>{ title }</p>
    </div>
  )
}
