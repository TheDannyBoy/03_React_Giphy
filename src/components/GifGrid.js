import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import getGifs from '../helpers/getGifs'

export const GifGrid = ({ category }) => {
  useEffect(() => {
    getGifs(category).then(setImagesList);
  }, [category]);

  const [imagesList, setImagesList] = useState([]);

  return (
    <>
      <h3>{ category }</h3>
      <div className='card-grid'>
        {
          imagesList.map(image => (
            <GifGridItem key={image.id} {...image} />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string,
}
