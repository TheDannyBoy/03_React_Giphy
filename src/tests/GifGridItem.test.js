import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';

describe('Testing GifGridItem', () => {
  const title = 'Soy un tÃ­tulo';
  const url = 'https://foo';

  test('Should create GifGridItem with title and url correctly', () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  })

  test('Should have a paragraph with the title', () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper.find('p').text().trim()).toBe(title);
  })

  test('Should have an image with the url as src and title as alt', () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    const imageProps = wrapper.find('img').props();

    expect(imageProps.src).toBe(url);
    expect(imageProps.alt).toBe(title);
  })

  test('The component should have animate__rubberBand as class', () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    const divProps = wrapper.find('div').prop('className');

    expect(divProps.includes('animate__rubberBand')).toBe(true);
  })
  
});
