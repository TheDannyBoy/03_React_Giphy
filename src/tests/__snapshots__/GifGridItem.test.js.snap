// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem Should create GifGridItem with title and url correctly 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="Soy un título"
    src="https://foo"
  />
  <p>
    Soy un título
  </p>
</div>
`;
